dataprofiler/tests/test_utils_global.py

```diff
@@ -0,0 +1,53 @@
+"""Validates that generator intakes DATAPROFILER_SEED properly."""
+import os
+import unittest
+import unittest.mock
+
+import numpy as np
+from numpy.random import PCG64
+
+from .. import utils_global
+
+
+class TestGetRandomNumberGenerator(unittest.TestCase):
+    """Validates get_random_number_generator() is properly working."""
+
+    @unittest.mock.patch.dict(os.environ, {"DATAPROFILER_SEED": "0"}, clear=True)
+    @unittest.mock.patch("dataprofiler.utils_global.settings._seed", new=123)
+    def test_dataprofiler_seed_true_settings_seed_false(self):
+        """Verify that we get the expected result when DATAPROFILER_SEED in os.environ and settings._seed!=None."""
+        with unittest.mock.patch("numpy.random.default_rng") as mock_fnc:
+            rng = utils_global.get_random_number_generator()
+            self.assertEqual(mock_fnc.call_count, 1)
```

Comment:
Instead of just call count, we should validate the value with something like `mock_fnc.assert_called_with(...)` we can then validate the last call received the value we expect.

Reply:
Currently, this doesn't ensure that when called, it uses the proper seed based on the settings \ env vars